.github/copilot-instructions.md

```diff
@@ -0,0 +1,61 @@
+This is a Go based repository hosting a Terrform provider for the elastic stack (elasticsearch and kibana) APIs.  This repo currently supports both [plugin framework](https://developer.hashicorp.com/terraform/plugin/framework/getting-started/code-walkthrough) and [sdkv2](https://developer.hashicorp.com/terraform/plugin/sdkv2) resources. All new resources _must_ use the plugin framework. 
+
+For further information, please see [README.md](../README.md) and the [CONTRIBUTING.md](../CONTRIBUTING.md) docs.
+
+## Code Standards
+
+### Required Before Each Commit
+- Run `make fmt` before committing any changes to ensure proper code formatting, this will run gofmt on all Go files to maintain consistent style.
+- Run `make lint` to ensure any linting errors have not surfaced with your changes
+
+### Required Before Pull Request
+- Run `make gen` to update the documentation and code based on your changes.
```

Comment:
```suggestion
- Run `make docs-generate` to update the documentation.
```

`make gen` is probably going to do things we don't want right now. Updating the Kibana client should be something we can support, but it's not there yet.